contracts/examples/LiquidityExamples.sol

```diff
@@ -0,0 +1,249 @@
+        // get liquidity data for tokenId
+        uint128 liquidity = deposits[tokenId].liquidity;
+        uint128 halfLiquidity = liquidity / 2;
+
+        // amount0Min and amount1Min are price slippage checks
+        // if the amount received after burning is not greater than these minimums, transaction will fail
+        INonfungiblePositionManager.DecreaseLiquidityParams memory params =
+            INonfungiblePositionManager.DecreaseLiquidityParams({
+                tokenId: tokenId,
+                liquidity: halfLiquidity,
+                amount0Min: 0,
+                amount1Min: 0,
+                deadline: block.timestamp
+            });
+
+        (amount0, amount1) = nonfungiblePositionManager.decreaseLiquidity(params);
+
+        //send liquidity back to owner
+        _sendToOwner(tokenId, amount0, amount1);
+    }
+
+    /// @notice Increases liquidity in the current range
+    /// @dev Pool must be initialized already to add liquidity
+    /// @param tokenId The id of the erc721 token
+    /// @param amount0 The amount to add of token0
+    /// @param amount1 The amount to add of token1
+    function increaseLiquidityCurrentRange(
+        uint256 tokenId,
+        uint256 amountAdd0,
+        uint256 amountAdd1
+    )
+        external
+        returns (
+            uint128 liquidity,
+            uint256 amount0,
+            uint256 amount1,
+            IUniswapV3Pool pool
+        )
+    {
+        AddLiquidityParams memory params =
+            AddLiquidityParams({
+                token0: deposits[tokenId].token0,
+                token1: deposits[tokenId].token1,
+                fee: poolFee,
+                recipient: address(this),
+                tickLower: deposits[tokenId].tickLower,
+                tickUpper: deposits[tokenId].tickUpper,
+                amount0Desired: amountAdd0,
+                amount1Desired: amountAdd1,
+                amount0Min: 0,
+                amount1Min: 0
+            });
+
+        (liquidity, amount0, amount1, pool) = addLiquidity(params);
+    }
+
+    /// @notice Transfers funds to owner of NFT
+    /// @param tokenId The id of the erc721
+    /// @param amount0 The amount of token0
+    /// @param amount1 The amount of token1
+    function _sendToOwner(
+        uint256 tokenId,
+        uint256 amount0,
+        uint256 amount1
+    ) internal {
+        // get owner of contract
+        address owner = deposits[tokenId].owner;
+
+        address token0 = deposits[tokenId].token0;
+        address token1 = deposits[tokenId].token1;
+        // send collected fees to owner
+        TransferHelper.safeTransfer(token0, owner, amount0);
+        TransferHelper.safeTransfer(token1, owner, amount1);
+    }
+
+    /// @notice Transfers the NFT to this contract
+    /// @dev The transfer triggers `onERC721Received`
+    /// @param tokenId The id of the erc721
+    function despositNFT(uint256 tokenId) external {
```

Comment:
Makes sense! Originally I had the logic of building the mapping of token -> owner in here but realized that should happen in the `onERC721Received` function. So now I see that this function is pretty useless since an external contract can just call transfer to this contract, triggering `onERC721Received`